### PR TITLE
Some cache change for filtering.

### PR DESCRIPTION
**Pull request by samuelgarcia**
* Make FilteredChunkCache optional.
  * name cache cache_on_file
  * avoid concatenate.

### CODE REVIEW

**Comment by samuelgarcia** [comment]
This is untested for now.
Just aproposal.
@alejoe91 this is for you.

**Comment by samuelgarcia** [comment]
There is something wear in the previsous code:
```
            ich1 = int(start_frame / self._chunk_size)
            ich2 = int((end_frame - 1) / self._chunk_size)
```
what happen if start_frame is not a multiple of chunksize ?

File: spiketoolkit/preprocessing/filterrecording.py [hunk 1]
```diff
@@ -46,8 +52,9 @@ def get_traces(self, channel_ids=None, start_frame=None, end_frame=None):
                 else:
                     end0 = self._chunk_size
                 chan_idx = [self.get_channel_ids().index(chan) for chan in channel_ids]
-                filtered_chunk_list.append(filtered_chunk0[chan_idx, start0:end0])
-            filtered_chunk =  np.concatenate(filtered_chunk_list, axis=1)
+                filtered_chunk[:, start0:end0] = filtered_chunk0[chan_idx, start0:end0]
+                # filtered_chunk_list.append(filtered_chunk0[chan_idx, start0:end0])
+            #�filtered_chunk =  np.concatenate(filtered_chunk_list, axis=1)
         else:
             filtered_chunk = self.filter_chunk(start_frame=0, end_frame=self._recording.get_num_frames())
```
Comment (alejoe91) [inline spiketoolkit/preprocessing/filterrecording.py]: This should be `filtered_chunk = self.filter_chunk(start_frame=0, end_frame=self._recording.get_num_frames())[channel_ids, start_frame:end_frame]` for consistency

Comment (samuelgarcia) [reply spiketoolkit/preprocessing/filterrecording.py]: Why not 
```filtered_chunk = self.filter_chunk(start_frame=start_frame:end_frame, end_frame=end_frame)[channel_ids, :]```

Comment (alejoe91) [reply spiketoolkit/preprocessing/filterrecording.py]: Also good

Comment (samuelgarcia) [reply spiketoolkit/preprocessing/filterrecording.py]: smater you want to say.

**Comment by samuelgarcia** [comment]
@alejoe91 : I think that this must be in the release.

**Comment by samuelgarcia** [comment]
This should be OK.
Alessio could you have a look ?

**Comment by alejoe91** [comment]
can you merge this with master? Cole changed some small things!

**Comment by alejoe91** [comment]
sorry I mean before we merge the PR! :)

**Comment by samuelgarcia** [comment]
@alejoe91 you can merge.
And change chunk_size back and also in spikeinterface.

**Pull request merged**